editor/src/clj/editor/settings_core.clj

```diff
@@ -38,22 +55,26 @@
   (BufferedReader. (StringReader. content)))
 
 (defn- empty-parse-state []
-  {:current-category nil :settings nil})
+  {:current-category nil :settings []})
 
 (defn- parse-category-line [{:keys [current-category settings] :as parse-state} line]
   (when-let [[_ new-category] (re-find #"^\s*\[([^\]]*)\]" line)]
     (assoc parse-state :current-category new-category)))
 
-(defn- parse-setting-line [{:keys [current-category settings] :as parse-state} line]
-  (when-let [[_ key val] (seq (map s/trim (re-find #"([^=]+)=(.*)" line)))]
-    (when-let [setting-path (seq (non-blank (s/split key #"\.")))]
-      (update parse-state :settings conj {:path (into [current-category] setting-path) :value val}))))
+ (defn- parse-setting-line [{:keys [current-category settings] :as parse-state} line]
+   (when-let [[_ key val] (seq (map s/trim (re-find #"([^=]+)=(.*)" line)))]
+     (when-let [setting-path (seq (non-blank (s/split key #"\.")))]
+       (let [settings (get-in parse-state [:settings])
+             full-path (seq (conj [current-category] (first setting-path)))]
+         (if-let [existing-value (get-setting settings full-path)]
+           (update parse-state :settings set-setting full-path (str existing-value "," val))
+           (update parse-state :settings conj {:path full-path :value val}))))))
 
 (defn- parse-error [line]
   (throw (Exception. (format "Invalid setting line: %s" line))))
 
 (defn- parse-state->settings [{:keys [settings]}]
-  (vec (reverse settings)))
+  settings)
```

Comment:
The Vector vs List thing above also has a different effect and that is that with a Vector we append to the end while a List always appends to the beginning. Previously when we operated on a List we had to reverse the List and convert it to a vector.